docs/source/advanced/training_tricks.rst

```diff
@@ -153,3 +153,46 @@ Advanced GPU Optimizations
 
 When training on single or multiple GPU machines, Lightning offers a host of advanced optimizations to improve throughput, memory efficiency, and model scaling.
 Refer to :doc:`Advanced GPU Optimized Training for more details <../advanced/advanced_gpu>`.
+
+----------
+
+Sharing Datasets Across Process Boundaries
+------------------------------------------
+The :class:`~pytorch_lightning.DataModule` class provides an organized way to decouple data loading from training logic, with :meth:`~pytorch_lightning.DataModule.prepare_data` being used for downloading and pre-processing the dataset on a single process, and :meth:`~pytorch_lightning.DataModule.setup` loading the pre-processed data for each process individually:
+
+.. code-block:: python
+
+    class MNISTDataModule(pl.LightningDataModule):
+        def prepare_data(self):
+            MNIST(self.data_dir, download=True)
+
+        def setup(self, stage: Optional[str] = None):
+            self.mnist = MNIST(self.data_dir)
+
+        def train_loader(self):
+            return DataLoader(self.mnist, batch_size=128)
+
+However, for in-memory datasets, that means that each process will hold a (redundant) replica of the dataset in memory, which may be impractical when using many processes and utilizing datasets that nearly fit into CPU memory, as the memory consumption will scale up linearly with the number of processes.
+For example, when training Graph Neural Networks, a common strategy is to load the entire graph into CPU memory for fast access to the entire graph structure and its features, and perform neighbor sampling to obtain mini-batches that fit onto the GPU.
+
+A simple way to disallow redundant dataset replicas is to rely on :obj:`torch.multiprocessing` to share the `data automatically between spawned processes via shared memory <https://pytorch.org/docs/stable/multiprocessing.html>`_.
```

Comment:
This can be a bit troublesome depending on how many individual tensors you have. Depending on the type of communication and the number of tensors this can also lead to errors with too many open file handles. Just wanted to mention this so that we don't advertise this as the best way at all circumstances.

Reply:
Sounds great. Thanks for the feedback. Let me add a note and modify the example according (i.e. sharing a single tensor instead of many small individual ones).

Reply:
Updated the doc according to your suggestions. Hope this resolve your concerns :)